nens_gs_uploader.py

```diff
@@ -0,0 +1,374 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 24 15:39:13 2019
+
+@author: chris.kerklaan - N&S
+
+TODOs:
+    1. Code review
+    2. Black voor formatten van scripts
+    3. Extra checks op shapefiles 
+    4. Add systematics for project
+    5. Standaard sld's 
+    6. Check slds werken nog niet goed
+    
+    
+"""
+# system imports
+import os
+import sys
+import logging
+from time import sleep
+from configparser import RawConfigParser
+
+# Third-party imports
+import ogr
+import gdal
+import argparse
+from glob import glob
+
+# Local imports
+from nens_gs_uploader.postgis import (
+    SERVERS,
+    PG_DATABASE,
+    copy2pg_database,
+    add_metadata_pgdatabase,
+)
+from nens_gs_uploader.project import (
+    logger,
+    log_time,
+    percentage,
+    print_list,
+    print_dictionary,
+)
+
+from nens_gs_uploader.wrap import wrap_geoserver
+from nens_gs_uploader.vector import wrap_shape
+from nens_gs_uploader.sld import wrap_sld
+from nens_gs_uploader.sld import replace_sld_field_based_on_shape
+from nens_gs_uploader.upload_ready import correct
+
+# Globals
+DRIVER_OGR_SHP = ogr.GetDriverByName("ESRI Shapefile")
+DRIVER_OGR_MEM = ogr.GetDriverByName("Memory")
+
+FILE = os.path.dirname(os.path.abspath(__file__))
+SLD_PATH = os.path.join(FILE, "nens_gs_uploader", "sld")
+
+
+# Exceptions
+ogr.UseExceptions()
+gdal.UseExceptions()
+
+# GDAL configuration options
+gdal.SetConfigOption("CPL_DEBUG", "ON")
+gdal.SetConfigOption("CPL_CURL_VERBOSE", "ON")
+
+# Logging configuration options
+for handler in logging.root.handlers[:]:
+    logging.root.removeHandler(handler)
+
+
+def get_parser():
+    """ Return argument parser. """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("inifile", metavar="INIFILE", help="Settings voor inifile.")
+    return parser
+
+
+def get_subject_from_path(path):
+    return path.split("\\")[-1].split(".shp")[0]
+
+
+def get_standard_sld(sld_name):
+    return os.path.join(SLD_PATH, sld_name + ".sld")
+
+
+def set_log_config(location, name="log"):
+    path = os.path.join(location, name)
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        filename=path + ".log",
+        level=logging.DEBUG,
+    )
+    gdal.SetConfigOption("CPL_LOG_ERRORS", path + "_gdal_err.log")
+    gdal.SetConfigOption("CPL_LOG", path + "_gdal.log")
+
+
+class settings_object(object):
+    """Reads settings from inifile settings from command line tool"""
+
+    def __init__(self, ini_file=None):
+        if ini_file is not None:
+            config = RawConfigParser()
+            config.read(ini_file)
+            self.ini = ini_file
+            self.ini_location = os.path.dirname(ini_file)
+            self.config = config
+
+            for section in config.sections():
+                for key, value in config.items(section):
+                    if value == "True":
```

Comment:
Is dit nodig? Volgens mij kan je in een configparser zeggen dat iets een boolean is en dan krijg je volgens mij automatisch een boolean terug i.p.v. een string?